Add Hyper-V firewall powershell cmdlet documentation (#3602)
MicrosoftDocs/windows-powershell-docs: docset/winserver2022-ps/netsecurity/Get-NetFirewallHyperVRule.md

Retrieves Hyper-V firewall rules from the target computer.

## SYNTAX

### GetAll (Default)
```
Get-NetFirewallHyperVRule [-All] [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob] [<CommonParameters>] 
```

### ByName
```
Get-NetFirewallHyperVRule [-Name] <string[]> [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob]  [<CommonParameters>]
```

### ByDisplayName
```
Get-NetFirewallHyperVRule -DisplayName <string[]> [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob]  [<CommonParameters>]
```

### ByQuery
```
Get-NetFirewallHyperVRule [-Direction {Inbound | Outbound}] [-VMCreatorId <string[]>] [-Protocol <string[]>] [-Action {NotConfigured | Allow | Block}] [-Enabled {True | False}] [-EnforcementStatus {Unknown | OK | PartiallyEnforced | NoApplicablePorts | ParsingError | Error}] [-PolicyStoreSourceType {None | Local | GroupPolicy | Dynamic | Generated | Hardcoded | MDM | HostFirewallLocal | HostFirewallGroupPolicy | HostFirewallDynamic | HostFirewallMDM}] [-PolicyStore <string>] [-CimSession <CimSession[]>] [-ThrottleLimit <int>] [-AsJob] [<CommonParameters>]
```

## DESCRIPTION

> The **Get-NetFirewallHyperVRule** cmdlet returns the instances of the Hyper-V firewall rules that match the search parameters from the user. See the New-NetFirewallHyperVrule cmdlet for more information.
> 
> This cmdlet returns one or more Hyper-V firewall rules by specifying the Name parameter, the DisplayName parameter, or rule properties. The queried rules can be placed intro variables and piped to other cmdlets for further modifications.

Reviewer: ```suggestion

The **Get-NetFirewallHyperVRule** cmdlet returns the instances of the Hyper-V firewall rules that match the search parameters of the user. See the New-NetFirewallHyperVrule cmdlet for more information.

This cmdlet returns one or more Hyper-V firewall rules by specifying the Name parameter, the DisplayName parameter, or rule properties. The queried rules can be placed into variables and piped to other cmdlets for further modifications.

```

In addition to the rule parameters, a read-only field called 'PortStatuses' will be enumerated. A Hyper-V firewall rule is applied to individual Hyper-V firewall ports based on the rule parameters. This output displays all Hyper-V ports that this rule is currently applied to.

## EXAMPLES

### EXAMPLE 1
```
PS C:\> Get-NetFirewallHyperVRule -PolicyStore ActiveStore
```

> This retrieves all of the Hyper-V firewall rules in the active store, which is a collection of all the policy stores that apply to the computer. Running this cmdlet without specifying the policy store retrieves the persistent store.

Reviewer: ```suggestion
This example retrieves all the Hyper-V firewall rules in the active store, which is a collection of all the policy stores that apply to the computer. Running this cmdlet without specifying the policy store retrieves the persistent store.

```

### EXAMPLE 2
```
PS C:\> Get-NetFirewallHyperVRule -EnforcementStatus OK
```

> This retrieves all of the Hyper-V firewall rules with EnforcementStatus OK, which means that the rule is active and being enforced.

Reviewer: ```suggestion
This example retrieves all the Hyper-V firewall rules with EnforcementStatus OK, which means that the rule is active and being enforced.

```

### EXAMPLE 3
```
PS C:\> Get-NetFirewallHyperVRule -Name 'MyRuleId'
```

> This retrieves the Hyper-V firewall rule with the Name 'MyRuleId'

Reviewer: ```suggestion
This example retrieves the Hyper-V firewall rule with the Name 'MyRuleId'

```

Accept wildcard characters: False
```

### -Direction
Specifies that matching Hyper-V firewall rules of the indicated direction are retrieved. 
This parameter specifies which direction of traffic to match with this rule. 
The acceptable values for this parameter are: Inbound or Outbound.

```yaml
Type: Direction
Parameter Sets: (All)
Aliases: 
Accepted values: Inbound, Outbound
Required: False
Position: Named
Default value: Inbound
Accept pipeline input: False
Accept wildcard characters: False
```

### -DisplayName
Specifies that only matching Hyper-V firewall rules of the indicated display name are retrieved. Wildcard characters are accepted.

```yaml
Type: String
Parameter Sets: (All)
Aliases: 
Required: True
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Enabled
Specifies that matching Hyper-V firewall rules of the indicated state are retrieved.
This parameter specifies that the rule object is administratively enabled or administratively disabled.
The acceptable values for this parameter are:
- True: Specifies the rule is currently enabled. 
- False: Specifies the rule is currently disabled.

> Note, that the type of this parameter is not boolean, therefore `$true` and `$false` variables are not acceptable values here. Use "True" and "False" text strings instead.

Reviewer: ```suggestion
Note that the type of this parameter is not Boolean, therefore `$true` and `$false` variables are not acceptable values here. Use "True" and "False" text strings instead.

```

By default, the PersistentStore is queried.

```yaml
Type: String
Parameter Sets: (All)
Aliases: 

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -PolicyStoreSourceType
Specifies that Hyper-V firewall rules that match the indicated policy store source type are retrieved. 
This parameter value is automatically generated and should not be modified. 
The acceptable values for this parameter are:
- Local: The object originates from the local store. 
- MDM: The object originates from the MDM store.
- Dynamic: The object originates from the dynamic store.
- HostFirewallLocal: This object originates from the host windows firewall local store.
- HostFirewallGroupPolicy: This object originates from the host windows firewall group policy store.
- HostFirewallMDM: This object originates from the host windows firewall MDM store.

By default, the Local store is queried.

```yaml
Type: PolicyStoreType[]
Parameter Sets: ByQuery
Aliases: 
Accepted values: None, Local, MDM

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Protocol

> Specifies that only matching Hyper-V firewall rules with the indicated Protocol are retrieved.

Reviewer: ```suggestion
Specifies that only matching Hyper-V firewall rules with the indicated protocol are retrieved.

```

The acceptable values for this parameter are:
- Protocols by number:  0-255. 
- Protocols by name:  TCP, UDP, ICMPv4, or ICMPv6. 

```yaml
Type: String
Parameter Sets: (All)
Aliases: 
Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -ThrottleLimit
Specifies the maximum number of concurrent operations that can be established to run the cmdlet.

> If this parameter is omitted or a value of `0` is entered, then Windows PowerShell® calculates an optimum throttle limit for the cmdlet based on the number of CIM cmdlets that are running on the computer.

Reviewer: ```suggestion
If this parameter is omitted or a value of `0` is entered, Windows PowerShell calculates an optimum throttle limit for the cmdlet based on the number of CIM cmdlets that are running on the computer.

```

```yaml
Type: Int32
Parameter Sets: (All)
Aliases: 
Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -VMCreatorId
Specifies that only matching Hyper-V firewall rules with the specified VMCreatorId are retrieved.
The format for this value is a GUID enclosed in brackets: '{9E288F02-CE00-4D9E-BE2B-14CE463B0298}'.

```yaml
Type: String
Parameter Sets: (All)
Aliases: 
Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### CommonParameters
This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).

## INPUTS

### None

## OUTPUTS

### Microsoft.Management.Infrastructure.CimInstance#root\StandardCimv2\NetFirewallHyperVRule
The `Microsoft.Management.Infrastructure.CimInstance` object is a wrapper class that displays Windows Management Instrumentation (WMI) objects.

> The path after the pound sign (`#`) provides the namespace and class name for the underlying WMI object.

Reviewer: ```suggestion
The path after the pound sign (`£`) provides the namespace and class name for the underlying WMI object.

```
